azure-pipelines.yml

```diff
@@ -114,4 +120,111 @@ stages:
             displayName: "Publish WebAPI Artifact"
             inputs:
               PathtoPublish: '$(Build.ArtifactStagingDirectory)\ExchangeSetServiceWebAPI'
-              ArtifactName: ExchangeSetServiceWebAPI
\ No newline at end of file
+              ArtifactName: ExchangeSetServiceWebAPI
+
+          - task: PublishBuildArtifacts@1
+            displayName: "Publish terraform Artifacts"
+            inputs:
+              PathtoPublish: '$(Build.SourcesDirectory)\Deployment'
+              ArtifactName: terraformartifact
+
+  - stage: Devdeploy
+    displayName: "Devdeploy (inc terraform, webapp deploy)"
+    jobs:
+      - deployment: DevDeployApp
+        displayName: "Dev - deploy terraform and dotnet App"
+        environment: "Ess-Dev"
+        workspace:
+          clean: all
+        variables:
+          - group: "Exchange Set Service Terraform Dev"
+          - group: "ESS-Dev-Variables"
+          - name: DeploymentResourceGroupName
+            value: ""
+          - name: DeploymentStorageAccountName
+            value: ""
+          - name: "Environment"
+            value: "dev"
+        strategy:
+          runOnce:
+            deploy:
+              steps:
+                - template: Deployment/templates/continous-deployment.yml
+                  parameters:
+                    terraformapply: ${{ parameters.terraformapply }}
+
+                - task: DownloadPipelineArtifact@2
+                  displayName: "Download Functional test Artifact"
+                  inputs:
+                    source: "current"
+                    artifact: "functionaltests"
+                    path: $(Build.SourcesDirectory)\functionaltests
+
+                - task: FileTransform@1
+                  displayName: "File Transform: functionaltests"
+                  inputs:
+                    folderPath: '$(Build.SourcesDirectory)\functionaltests\UKHO.ExchangeSetService.API.FunctionalTests'
+                    fileType: json
+                    targetFiles: 'appsettings.json'
+
+                - task: DotNetCoreCLI@2
+                  displayName: "Run Functional tests"
+                  inputs:
+                    command: "test"
+                    projects: |
+                      **\*FunctionalTest*.dll
+                      !**\*TestAdapter.dll
+                      !**\obj\**
+                    testRunTitle: "Dev-AutomationTests"
+                    workingDirectory: '$(Build.SourcesDirectory)\functionaltests'
+
+  - stage: QAdeploy
+    displayName: QAdeploy (inc terraform, webapp deploy)
+    jobs:
+      - deployment: QADeployApp
+        displayName: QA - deploy terraform and dotnet App
+        condition: and(succeeded(), or(eq(variables['Build.SourceBranch'], 'refs/heads/main'),startsWith(variables['Build.SourceBranch'], 'refs/heads/release/')))
+        environment: "Ess-Qa"
+        workspace:
+          clean: all
+        variables:
+          - group: "Exchange Set Service Terraform Qa"
```

Comment:
```suggestion
          - group: "Exchange Set Service Terraform QA"
```

Reply:
This library is still not added by ddc I have kept just foreseeing it. Will replace it with the actual one once library is created.